transport/tchannel/header.go

```diff
@@ -61,10 +97,38 @@ func readHeaders(format tchannel.Format, getReader func() (tchannel.ArgReader, e
 	if headers == nil {
 		headers = make(transport.Headers)
 	}
-
 	return headers, r.Close()
 }
 
+func writeRequestHeaders(
+	ctx context.Context,
+	format tchannel.Format,
+	appheaders transport.Headers,
+	getWriter func() (tchannel.ArgWriter, error),
+) error {
+	ctxheaders := baggage.FromContext(ctx)
+	headers := make(transport.Headers, len(ctxheaders)+len(appheaders))
```

Comment:
this is fine for now, but I'd like to do a 0-alloc version of `writeRequestHeaders` that propagates both the header bags to the `writeHeaders` call, at least for the Thrift encoding. That will avoid the allocs + cost of building a new map which we discard almost immediately